geoh5py/objects/cell_object.py

```diff
@@ -118,11 +118,6 @@ def remove_cells(self, indices: list[int] | np.ndarray, clear_cache: bool = Fals
         :param indices: Indices of cells to be removed.
         :param clear_cache: Clear cache of data values.
         """
-
-        if self.cells is None:
-            warnings.warn("No cells to be removed.", UserWarning)
-            return
-
```

Comment:
Keep it.

Reply:
Technically, if the cell is None, it will be loaded from the file. 
So they are no chance cell being None! even with clear_cache,
Because you are reloading cells.
And you can't have a cell object with empty cells now!